python/tvm/topi/image/resize.py

```diff
@@ -59,6 +59,33 @@ def get_2d_pixel(data, layout, boxes, image_height, image_width, n, c, y, x, cc,
     return data(n, c, y, x, cc).astype("float")
 
 
+def get_iny_inx(
+    y, x, image_height, image_width, target_height, target_width, coordinate_transformation_mode
+):
+    scale_y = te.div(image_height.astype("float"), target_height.astype("float"))
+    scale_x = te.div(image_width.astype("float"), target_width.astype("float"))
+    if coordinate_transformation_mode == "half_pixel":
+        in_y = (y + 0.5) * scale_y - 0.5
+        in_x = (x + 0.5) * scale_x - 0.5
+    elif coordinate_transformation_mode == "align_corners":
+        in_y = y * (image_height - 1).astype("float") / (target_height - 1)
+        in_x = x * (image_width - 1).astype("float") / (target_width - 1)
+    elif coordinate_transformation_mode == "asymmetric":
+        in_y = y * scale_y
+        in_x = x * scale_x
+    elif coordinate_transformation_mode == "pytorch_half_pixel":
```

Comment:
Do we need to update the pytorch and tf frontends to use these new modes?

Reply:
I'm honestly not sure what the frameworks are expecting or what tests they currently do or do not pass?

Reply:
Hmm I thought we were testing `half_pixel` in PT frontend but indeed it doesn't seem so. I think `half_pixel` corresponds to `bilinear` + `align_corners=False` but this combination is not there below

https://github.com/apache/tvm/blob/b24fbe7dee10195b6585634c0625828fe8624d5f/tests/python/frontend/pytorch/test_forward.py#L1701-L1703

Reply:
Tried adding `verify_model(Upsample(size=(50, 50), mode="bilinear", align_corners=False), inp)` to the test, it worked fortunately.

Reply:
Also when I added `half_pixel` option, I intended this option to correspond exactly to ONNX `pytorch_half_pixel`. In PT ONNX exporter, ``pytorch_half_pixel` is introduced for bilinear + align_corners=False case, and that is also when `half_pixel` is used in our PT frontend. See https://github.com/pytorch/pytorch/blob/c371542efc31b1abfe6f388042aa3ab0cef935f2/torch/onnx/symbolic_helper.py#L517-L518

So we probably don't need ``pytorch_half_pixel`. We can update `half_pixel` if necessary.

Reply:
The ONNX tests definitely have some failures if I use half_pixel when ONNX suggests pytorch_half_pixel, the issue is the if_then_else forcing some values to 0

Reply:
Ok I looked at the spec again, and indeed I realized they have two variants of `half_pixel`... I don't know what `pytorch_half_pixel` is for, probably only for the case where the resized shape is 1?

Reply:
No, the only test ONNX currently has with a target shape 1 is the `pytorch_half_pixel` test. I separated them based on the ONNX Operator documentation and the ORT implementation, but it doesn't look like I have a test that hits the difference with a `half_pixel` mode.

Reply:
I would prefer to leave the separation for that future use case that's assuming `half_pixel` instead of `pytorch_half_pixel` for a resize to 1, even if we don't see it yet.